### PR TITLE
feat: replace tool table with a tool grid

### PR DESCRIPTION
**Pull request by tylerslaton**
https://github.com/user-attachments/assets/40b0ff3f-a792-4c53-b5cc-09d3ba5e9c76

### CODE REVIEW

File: ui/admin/app/components/TruncatedText.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    maxWidth,
+}: {
+    content: string;
+    maxWidth: string;
+}) {
+    return (
+        <TooltipProvider>
+            <Tooltip>
+                <TooltipTrigger asChild>
+                    <div className={`${maxWidth} truncate cursor-pointer`}>
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/TruncatedText.tsx:20]: does this already add a `...` to the end of the text?

File: ui/admin/app/components/tools/toolGrid/ToolCard.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    {tool.error && (
+                        <TooltipProvider>
+                            <Tooltip>
+                                <TooltipTrigger>
+                                    <Badge className="ml-2 bg-error mb-1">
+                                        Failed
+                                    </Badge>
+                                </TooltipTrigger>
+                                <TooltipContent className="max-w-xs bg-error-foreground border border-error text-foreground">
+                                    <p>{tool.error}</p>
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/toolGrid/ToolCard.tsx]: missed Typography here

File: ui/admin/app/components/tools/toolGrid/ToolGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { CategoryTools } from "~/components/tools/toolGrid/CategoryTools";
+
+interface ToolGridProps {
+    tools: ToolReference[];
+    filter: string;
+    onDelete: (id: string) => void;
+}
+
+export function ToolGrid({ tools, filter, onDelete }: ToolGridProps) {
+    const filteredTools = tools?.filter(
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/tools/toolGrid/ToolGrid.tsx]: nit: can you `useMemo` these more expensive calculations?

File: ui/admin/app/components/tools/toolGrid/ToolCard.tsx (hunk 2 above)
Comment (tylerslaton) [reply ui/admin/app/components/tools/toolGrid/ToolCard.tsx]: Oops, thanks.

File: ui/admin/app/components/tools/toolGrid/ToolGrid.tsx (hunk 3 above)
Comment (tylerslaton) [reply ui/admin/app/components/tools/toolGrid/ToolGrid.tsx]: Done!

**Pull request merged**